mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts

```diff
@@ -15,9 +17,50 @@ export class WayPreviewDAL {
    */
   public static async getWaysPreview(): Promise<WayPreview[]> {
     const waysDTO = await WayService.getWaysDTO();
-    const waysUuids = waysDTO.map((item) => item.uuid);
 
-    const waysPreview = await Promise.all(waysUuids.map(WayPreviewDAL.getWayPreview));
+    const allNeededUsersUuids = new Set(waysDTO.flatMap(wayDTO => [
+      wayDTO.ownerUuid,
+      ...wayDTO.mentorUuids,
+      ...wayDTO.mentorRequestUuids,
+    ]));
+
+    const allNeededGoalsUuids = new Set(waysDTO.flatMap(wayDTO => [wayDTO.goalUuid]));
+
+    const allNeededUsersPreview = await UserPreviewDAL.getUsersPreviewByUuids(Array.from(allNeededUsersUuids));
+
+    const allNeededGoalsPreview = await GoalPreviewDAL.getGoalsPreviewByUuids(Array.from(allNeededGoalsUuids));
+
+    const waysPreview = await Promise.all(waysDTO.map(async (wayDTO) => {
+      const owner = allNeededUsersPreview.find(user => user.uuid === wayDTO.ownerUuid);
+      const mentors = wayDTO.mentorUuids.map(uuid => allNeededUsersPreview.find(user => user.uuid === uuid));
+      const mentorRequests = wayDTO.mentorRequestUuids.map(uuid => allNeededUsersPreview.find(user => user.uuid === uuid));
+      const wayGoal = allNeededGoalsPreview.find(goal => goal.uuid === wayDTO.goalUuid);
+
+      if (!owner) {
+        throw new Error(`Owner not found for UUID: ${wayDTO.ownerUuid}, WayUuid: ${wayDTO.uuid}`);
+      }
```

Comment:
there is no sence to check existense in each layer, could we avoid it? Zod already checked as I understand

Reply:
Removed